app/controllers/course/material/folders_controller.rb

```diff
@@ -0,0 +1,52 @@
+class Course::Material::FoldersController < Course::ComponentController
+  load_and_authorize_resource :material_folder, through: :course, parent: false,
+                                                class: Course::Material::Folder.name
+
+  def show
+    @subfolders = @material_folder.children
+  end
+
+  def new_subfolder
+    @subfolder = Course::Material::Folder.new
+  end
+
+  def create
+    parent_folder = @material_folder.parent
+    if @material_folder.save
+      redirect_to course_material_folder_path(current_course, parent_folder),
+                  success: t('.success', name: @material_folder.name)
+    else
+      redirect_to new_subfolder_course_material_folder_path(current_course, parent_folder),
```

Comment:
In new action, parent folder is `@material_folder`, that's issue of preserving `create` action